File: common_business/src/main/java/com/xiaoyv/busines/room/entity/SshEntity.java
package com.xiaoyv.busines.room.entity;

import androidx.annotation.Nullable;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

import com.blankj.utilcode.util.TimeUtils;
import com.xiaoyv.busines.config.SshLoginType;
import com.xiaoyv.busines.room.database.DateBaseConfig;

import java.io.Serializable;

/**
 * RdpEntity
 *
 * @author why
 * @since 2020/11/29
 **/
@Entity(tableName = DateBaseConfig.TABLE_SSH, indices = {@Index(value = {"ip"}, unique = true)})
public class SshEntity implements Serializable {
    private static final long serialVersionUID = 8962195813711670211L;
    @PrimaryKey(autoGenerate = true)
    public int id;
    @SshLoginType
    public int authType = SshLoginType.TYPE_PASSWORD;
    public long lastTime;
    public String label;
    public String group;
    public String ip;
    public String port;
    public String account;
    public String password;
    public String domain;
    public String setting;

    public SshEntity() {
        this.lastTime = TimeUtils.getNowMills();
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return super.equals(obj);
    }
}

File: common_business/src/main/java/com/xiaoyv/busines/room/entity/SshEntity.kt
package com.xiaoyv.busines.room.entity

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.blankj.utilcode.util.TimeUtils
import com.xiaoyv.busines.config.SshLoginType
import com.xiaoyv.busines.room.database.DateBaseConfig
import java.io.Serializable

/**
 * RdpEntity
 *
 * @author why
 * @since 2020/11/29
 */
@Entity(tableName = DateBaseConfig.TABLE_SSH, indices = [Index(value = ["ip"], unique = true)])
class SshEntity : Serializable {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @JvmField
    var lastTime: Long = TimeUtils.getNowMills()

    @JvmField
    @SshLoginType
    var authType = SshLoginType.TYPE_PASSWORD

    @JvmField
    var label: String = ""

    @JvmField
    var group: String = ""

    @JvmField
    var ip: String = ""

    @JvmField
    var port: String = ""

    @JvmField
    var account: String = ""

    @JvmField
    var password: String = ""

    @JvmField
    var domain: String = ""

    @JvmField
    var setting: String = ""
}